lib/popcorn/pop_templates.nit

```diff
@@ -43,4 +43,29 @@ redef class HttpResponse
 		var tpl = new TemplateString.from_file(file)
 		template(tpl, values, status)
 	end
+
+	# Render `pug_string` with the command cli `pug` and use data from `json`
+	#
+	# See https://pugjs.org/api/getting-started.html for more details on pug.
+	fun pug(pug_string: nullable String, json: nullable Serializable, status: nullable Int) do
+		var process
+		if json == null then
+			process = new ProcessDuplex("pug", "-D")
+		else
+			process = new ProcessDuplex("pug", "-D", "-O", json.to_json)
+		end
+		process.write pug_string or else ""
```

Comment:
you should use `ProcessDuplex::write_and_read` that is more robust (or indented to)